### PR TITLE
Cleaning servers artefacts in test suite after test execution

### PR DESCRIPTION
**Pull request by ehsavoie**
Removing the tests artefacts to reduce the filesystem usage of the test suite

### CODE REVIEW

**Review by rhusar — CHANGES_REQUESTED** [review]
-1

These are used to investigate test failures after the tests run and this doesn't even seem to be opt-in.

**Comment by pferraro** [comment]
:-1: This execution should not be enabled by default.  The general concept is fine (e.g. for CI runs), but IMO, it should be explicitly enabled via a property.

**Comment by ehsavoie** [comment]
In case of failure in the test the server directories wouldn't be deleted except if using -fae as the build would stop before the clean is called so I didn't think opt-in was required.
Adding a profile for this cleanup

File: testsuite/domain/pom.xml [hunk 1]
```diff
@@ -305,6 +305,41 @@
                 </plugins>
             </build>
         </profile>
+
+        <profile>
+            <id>ci-cleanup</id>
+            <activation>
+                <property>
+                    <name>cleanup</name>
```
Comment (bstansberry) [inline testsuite/domain/pom.xml]: How about ci-cleanup? The term "cleanup" is pretty generic.

File: testsuite/domain/pom.xml [hunk 2]
```diff
@@ -305,6 +305,41 @@
...
+            </activation>
+            <build>
+                <plugins>
+                    <!-- Clean the server configuration(s) -->
+                    <plugin>
+                        <groupId>org.apache.maven.plugins</groupId>
+                        <artifactId>maven-clean-plugin</artifactId>
+                        <executions>
+                            <execution>
+                                <id>clean-manual-mode-servers</id>
```
Comment (bstansberry) [inline testsuite/domain/pom.xml]: This id is repeated in all the poms, so if it's not going to be unique per pom it shouldn't mention manual-mode.

**Comment by bstansberry** [comment]
Our CI jobs all use --fae, so we can't count on this not running in the case of CI failure. And in the case of failure we can't be deleting the log files. The reason we archive log files in CI (for example see https://ci.wildfly.org/viewLog.html?buildId=52414&buildTypeId=WF_MasterLinux&tab=artifacts) is to make them available when analyzing failed runs.

**Comment by bstansberry** [comment]
@ehsavoie Adding this to each fileset should address my concern about the logs:

```
          <excludes>
            <exclude>**/*.log</exclude>
          </excludes>
```

**Comment by ehsavoie** [comment]
retest this please

**Pull request closed without merging**